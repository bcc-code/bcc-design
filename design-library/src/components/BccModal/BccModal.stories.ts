import BccModal from "./BccModal.vue";
import BccButton from "../BccButton/BccButton.vue";
import { CheckCircleFillIcon } from "@bcc-code/icons-vue";

import type { Meta, StoryFn } from "@storybook/vue3";

/**
 * A versatile modal component. Be sure to read the [guidelines](https://bccit.notion.site/Modals-e79ddadb127d4ff6aa6650379132d09a) on how to use this component.
 */
export default {
  title: "Components/BccModal",
  component: BccModal,
  argTypes: {},
} as Meta<typeof BccModal>;

const Template: StoryFn<typeof BccModal> = (args) => ({
  components: { BccModal, BccButton },
  setup() {
    return { args };
  },
  template: `
    <BccModal v-bind="args" @close="args.open = false">
        <template #default>
          <span v-if="args.slotDefault">
            {{ args.slotDefault }}
          </span>
          <span v-else>
            Is your email address <strong>lida@example.org</strong>?
          </span>
        </template>

        <template #secondaryAction>
          <BccButton variant="secondary" @click="args.open = false">
            {{ args.secondaryActionText || "Cancel" }}
          </BccButton>
        </template>

        <template #primaryAction>
          <BccButton @click="args.open = false">
            {{ args.primaryActionText || "Confirm" }}
          </BccButton>
        </template>
    </BccModal>

    <BccButton variant="secondary" @click="args.open=true">Open modal</BccButton>
  `,
});

/**
 * Control the content of the modal with the `title` prop and the `default` slot. Pass buttons to the `primaryAction` and `secondaryAction` slots. Set `closeButton` to `false` to hide the close button on desktops. The modal emits a `close` event when the user clicks the close button or closes the modal by clicking outside of it or pressing Escape.
 */
export const Example = Template.bind({});
Example.args = {
  open: false,
  title: "Confirm email address",
  closeButton: true,
  slotDefault: "",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccModal title="Confirm email address" :open="showModal" @close="showModal = false">
  Is your email address <strong>lida@example.org</strong>?

  <template #secondaryAction>
    <BccButton variant="secondary" @click="showModal = false">Cancel</BccButton>
  </template>

  <template #primaryAction>
    <BccButton @click="showModal = false">Confirm</BccButton>
  </template>
</BccModal>
    `,
    },
  },
};

const ExtraSlotsTemplate: StoryFn<typeof BccModal> = (args) => ({
  components: { BccModal, BccButton, CheckCircleFillIcon },
  setup() {
    return { args };
  },
  template: `
    <BccModal v-bind="args" @close="args.open = false">
      <template #header>
        <div class="text-heading-base flex items-center gap-2 text-success pb-2">
          <CheckCircleFillIcon class="w-4 h-4" /> Success!
        </div>

        <p class="text-secondary">
          You successfully registered to activity. Don't forget to register for activities for Day 2 & Day 3
        </p>
      </template>

        <template #default>
          You can join the activity chat by clicking the link below:
        </template>

        <template #secondaryAction>
          <BccButton variant="secondary" @click="args.open = false">
            Close
          </BccButton>
        </template>

        <template #primaryAction>
          <BccButton @click="args.open = false">
            Join activity chat
          </BccButton>
        </template>
    </BccModal>

    <BccButton variant="secondary" @click="args.open=true">Open modal</BccButton>
  `,
});

/**
 * Pass content to the `header` or `footer` slots to render extra content above and below the normal modal content.
 */
export const ExtraSlots = ExtraSlotsTemplate.bind({});
ExtraSlots.args = {
  open: false,
  closeButton: true,
  title: "Join the activity chat group",
  primaryActionText: "Open Telegram chat",
  secondaryActionText: "Close",
};
ExtraSlots.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccModal
  title="Join the activity chat group"
  :open="showModal"
  @close="showModal = false"
>
  <template #header>
    <div class="text-heading-base flex items-center gap-2 text-success pb-2">
      <CheckCircleFillIcon class="w-4 h-4" /> Success!
    </div>

    <p class="text-secondary">
      You successfully registered to activity. Don't forget to register for activities for Day 2 & Day 3
    </p>
  </template>

  <template #default>
    You can join the activity chat by clicking the link below:
  </template>

  <template #secondaryAction>
    <BccButton variant="secondary" @click="showModal = false">Cancel</BccButton>
  </template>

  <template #primaryAction>
    <BccButton @click="showModal = false">Confirm</BccButton>
  </template>
</BccModal>
    `,
    },
  },
};

const ScrollableTemplate: StoryFn<typeof BccModal> = (args) => ({
  components: { BccModal, BccButton },
  setup() {
    return { args };
  },
  template: `
    <BccModal v-bind="args" @close="args.open = false" class="max-w-[80vw]">
        <template #default>
          <p class="mb-4">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam id mattis sapien. Morbi euismod convallis tellus, in rutrum risus. Vivamus quam dui, auctor sit amet nibh non, dignissim volutpat massa. Nunc accumsan mauris vel congue finibus. Nulla venenatis maximus libero, quis imperdiet justo mattis sit amet. Vestibulum a odio a arcu accumsan malesuada. Mauris porttitor egestas orci nec bibendum. Sed accumsan, erat nec cursus vehicula, nibh risus porta tellus, eget rhoncus erat sem at nunc. Aliquam ornare, nibh blandit molestie tincidunt, ligula lectus faucibus diam, sed tristique arcu neque interdum purus. Morbi a blandit eros, quis suscipit sapien. Donec dictum dignissim maximus. Sed ac eros consectetur, ultrices dui a, rhoncus mi.
          </p>
          <p class="mb-4">
            In eu diam nisl. Maecenas urna tortor, hendrerit in varius ut, laoreet sit amet lectus. Phasellus vel massa vel nunc efficitur dapibus a et ipsum. Proin placerat lacus porttitor vulputate scelerisque. Sed sed lacinia nibh. Sed convallis feugiat malesuada. Nunc ullamcorper sit amet sapien vel pulvinar.
          </p>

          <p class="mb-4">
            Etiam ullamcorper posuere libero tempor hendrerit. Donec faucibus quis nunc non vehicula. In sit amet nibh interdum, hendrerit sem a, elementum est. Aliquam viverra tristique mi, ac ultricies orci congue vel. Nam imperdiet sem non efficitur porttitor. Phasellus ac nunc euismod est porta tincidunt. Pellentesque dapibus tortor ex, sed laoreet nunc mattis finibus. Donec sollicitudin neque a leo fermentum, quis maximus ex bibendum. Nulla dictum nisl nec facilisis venenatis. Mauris imperdiet vitae nulla eget euismod. Aenean dictum rutrum dui, cursus blandit lectus ultrices eget. Morbi non tristique elit, a tempor lectus. Nullam quis bibendum orci, et ornare eros. Nunc commodo dolor leo. In tincidunt imperdiet arcu, nec pharetra diam feugiat pulvinar. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
          </p>

          <p class="mb-4">
            Quisque at laoreet leo. Quisque eu nulla consequat, viverra enim tristique, congue libero. Fusce et gravida nisi. Ut consectetur lectus massa, sit amet gravida eros tempus in. Suspendisse suscipit orci bibendum rhoncus venenatis. Pellentesque varius luctus volutpat. Praesent velit urna, bibendum sed lacinia ut, hendrerit ac risus. Sed elementum congue dui, quis bibendum justo tempus sed. Integer vel sem id elit dignissim laoreet in id tellus. Morbi vehicula sagittis mattis. Nam cursus, diam sit amet iaculis egestas, metus dolor mollis libero, sit amet ornare sapien massa at neque. Suspendisse mollis nisl massa, vitae sagittis lorem condimentum non.
          </p>

          <p class="mb-4">
            Nulla venenatis nisi eu pulvinar accumsan. Vivamus condimentum lectus sed ligula euismod molestie. Aliquam facilisis malesuada leo, quis gravida ligula. Etiam lorem orci, auctor quis rutrum interdum, eleifend sit amet nulla. Morbi fermentum mi in lacinia aliquet. Quisque accumsan orci urna, id consequat sem ullamcorper in. Phasellus sodales mattis pharetra. Nam non mauris eget sapien efficitur pulvinar. Suspendisse potenti. Aenean vitae porta libero, quis malesuada urna. Aliquam gravida sollicitudin elit a lacinia. Nullam non est lacus. Donec posuere massa eleifend eros lobortis rhoncus.  
          </p>

          <p class="mb-4">
            Etiam ullamcorper posuere libero tempor hendrerit. Donec faucibus quis nunc non vehicula. In sit amet nibh interdum, hendrerit sem a, elementum est. Aliquam viverra tristique mi, ac ultricies orci congue vel. Nam imperdiet sem non efficitur porttitor. Phasellus ac nunc euismod est porta tincidunt. Pellentesque dapibus tortor ex, sed laoreet nunc mattis finibus. Donec sollicitudin neque a leo fermentum, quis maximus ex bibendum. Nulla dictum nisl nec facilisis venenatis. Mauris imperdiet vitae nulla eget euismod. Aenean dictum rutrum dui, cursus blandit lectus ultrices eget. Morbi non tristique elit, a tempor lectus. Nullam quis bibendum orci, et ornare eros. Nunc commodo dolor leo. In tincidunt imperdiet arcu, nec pharetra diam feugiat pulvinar. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
          </p>

          <p class="mb-4">
            Quisque at laoreet leo. Quisque eu nulla consequat, viverra enim tristique, congue libero. Fusce et gravida nisi. Ut consectetur lectus massa, sit amet gravida eros tempus in. Suspendisse suscipit orci bibendum rhoncus venenatis. Pellentesque varius luctus volutpat. Praesent velit urna, bibendum sed lacinia ut, hendrerit ac risus. Sed elementum congue dui, quis bibendum justo tempus sed. Integer vel sem id elit dignissim laoreet in id tellus. Morbi vehicula sagittis mattis. Nam cursus, diam sit amet iaculis egestas, metus dolor mollis libero, sit amet ornare sapien massa at neque. Suspendisse mollis nisl massa, vitae sagittis lorem condimentum non.
          </p>

          <p class="mb-4">
            Nulla venenatis nisi eu pulvinar accumsan. Vivamus condimentum lectus sed ligula euismod molestie. Aliquam facilisis malesuada leo, quis gravida ligula. Etiam lorem orci, auctor quis rutrum interdum, eleifend sit amet nulla. Morbi fermentum mi in lacinia aliquet. Quisque accumsan orci urna, id consequat sem ullamcorper in. Phasellus sodales mattis pharetra. Nam non mauris eget sapien efficitur pulvinar. Suspendisse potenti. Aenean vitae porta libero, quis malesuada urna. Aliquam gravida sollicitudin elit a lacinia. Nullam non est lacus. Donec posuere massa eleifend eros lobortis rhoncus.  
          </p>

          <p class="mb-4">
            Etiam ullamcorper posuere libero tempor hendrerit. Donec faucibus quis nunc non vehicula. In sit amet nibh interdum, hendrerit sem a, elementum est. Aliquam viverra tristique mi, ac ultricies orci congue vel. Nam imperdiet sem non efficitur porttitor. Phasellus ac nunc euismod est porta tincidunt. Pellentesque dapibus tortor ex, sed laoreet nunc mattis finibus. Donec sollicitudin neque a leo fermentum, quis maximus ex bibendum. Nulla dictum nisl nec facilisis venenatis. Mauris imperdiet vitae nulla eget euismod. Aenean dictum rutrum dui, cursus blandit lectus ultrices eget. Morbi non tristique elit, a tempor lectus. Nullam quis bibendum orci, et ornare eros. Nunc commodo dolor leo. In tincidunt imperdiet arcu, nec pharetra diam feugiat pulvinar. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
          </p>

          <p class="mb-4">
            Quisque at laoreet leo. Quisque eu nulla consequat, viverra enim tristique, congue libero. Fusce et gravida nisi. Ut consectetur lectus massa, sit amet gravida eros tempus in. Suspendisse suscipit orci bibendum rhoncus venenatis. Pellentesque varius luctus volutpat. Praesent velit urna, bibendum sed lacinia ut, hendrerit ac risus. Sed elementum congue dui, quis bibendum justo tempus sed. Integer vel sem id elit dignissim laoreet in id tellus. Morbi vehicula sagittis mattis. Nam cursus, diam sit amet iaculis egestas, metus dolor mollis libero, sit amet ornare sapien massa at neque. Suspendisse mollis nisl massa, vitae sagittis lorem condimentum non.
          </p>

          <p class="mb-4">
            Nulla venenatis nisi eu pulvinar accumsan. Vivamus condimentum lectus sed ligula euismod molestie. Aliquam facilisis malesuada leo, quis gravida ligula. Etiam lorem orci, auctor quis rutrum interdum, eleifend sit amet nulla. Morbi fermentum mi in lacinia aliquet. Quisque accumsan orci urna, id consequat sem ullamcorper in. Phasellus sodales mattis pharetra. Nam non mauris eget sapien efficitur pulvinar. Suspendisse potenti. Aenean vitae porta libero, quis malesuada urna. Aliquam gravida sollicitudin elit a lacinia. Nullam non est lacus. Donec posuere massa eleifend eros lobortis rhoncus.  
          </p>
        </template>

        <template #footer>
          <label class="flex items-center gap-2">
            <input type="checkbox"> I agree lorem ipsum dolores
          </label>
        </template>

        <template #secondaryAction>
          <BccButton variant="secondary" @click="args.open = false">
            Take me back
          </BccButton>
        </template>

        <template #primaryAction>
          <BccButton @click="args.open = false">
            Confirm
          </BccButton>
        </template>
    </BccModal>

    <BccButton variant="secondary" @click="args.open=true">Open modal</BccButton>
  `,
});

/**
 * The content of the modal will be scrollable if there is too much to show at once, with the header, actions and `footer` slot always being in view. Use a `class` or `style` attribute to make the modal larger than the provided `size`s if that is needed for you
 */
export const Scrollable = ScrollableTemplate.bind({});
Scrollable.args = {
  open: false,
  title: "Heading",
};
Scrollable.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccModal
  title="Heading"
  :open="showModal"
  @close="showModal = false"
  class="max-w-[80vw]"
>
  Lorem ipsum dolor sit amet...
  
  <template #footer>
    <label class="flex items-center gap-2">
      <input type="checkbox"> I agree lorem ipsum dolores
    </label>
  </template>

  <template #secondaryAction>
    <BccButton variant="secondary" @click="showModal = false">Take me back</BccButton>
  </template>

  <template #primaryAction>
    <BccButton @click="showModal = false">Confirm</BccButton>
  </template>
</BccModal>
    `,
    },
  },
};
