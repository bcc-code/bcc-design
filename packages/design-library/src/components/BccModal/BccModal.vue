<script setup lang="ts">
import { Dialog, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from "@headlessui/vue";
import { CloseIcon } from "@bcc-code/icons-vue";
import BccButton from "../BccButton/BccButton.vue";
import { computed, useSlots } from "vue";

// eslint-disable-next-line no-undef
defineOptions({
  inheritAttrs: false,
});

type Props = {
  open: boolean;
  title?: string;
  closeButton?: boolean;
};

const props = withDefaults(defineProps<Props>(), {
  open: false,
  closeButton: true,
});

const emit = defineEmits(["close"]);

const slots = useSlots();
const showCloseButton = computed(() => props.closeButton && !slots.header);
</script>

<template>
  <TransitionRoot as="template" :show="open">
    <Dialog as="div" class="bcc-modal-overlay-wrapper" @close="emit('close')">
      <div class="flex h-screen w-screen items-center justify-center overflow-hidden">
        <TransitionChild
          as="div"
          enter="ease-out duration-300"
          enter-from="opacity-0"
          enter-to="opacity-100"
          leave="ease-in duration-200"
          leave-from="opacity-100"
          leave-to="opacity-0"
        >
          <div class="bcc-modal-overlay" />
        </TransitionChild>

        <TransitionChild
          as="template"
          enter="ease-out duration-300"
          enter-from="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
          enter-to="opacity-100 translate-y-0 sm:scale-100"
          leave="ease-in duration-200"
          leave-from="opacity-100 translate-y-0 sm:scale-100"
          leave-to="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
        >
          <div class="bcc-modal-wrapper">
            <DialogPanel class="bcc-modal" v-bind="$attrs">
              <div class="bcc-modal-header" v-if="$slots.header">
                <slot name="header" />
              </div>

              <div class="bcc-modal-body" v-if="title || showCloseButton || slots.default">
                <div class="bcc-modal-title" v-if="title || showCloseButton">
                  <DialogTitle as="h3">{{ title }}</DialogTitle>
                  <button
                    @click.prevent="emit('close')"
                    v-if="showCloseButton"
                    class="bcc-modal-close-button"
                    aria-label="Close modal window"
                  >
                    <CloseIcon class="bcc-modal-close-icon" />
                  </button>
                </div>

                <div class="bcc-modal-content">
                  <slot />
                </div>
              </div>

              <div class="bcc-modal-footer">
                <div class="bcc-modal-footer-content" v-if="$slots.footer">
                  <slot name="footer" />
                </div>

                <div class="bcc-modal-actions">
                  <div class="bcc-modal-secondary-action" v-if="$slots.secondaryAction">
                    <slot name="secondaryAction"></slot>
                  </div>
                  <div class="bcc-modal-primary-action">
                    <slot name="primaryAction">
                      <BccButton @click="emit('close')"> OK </BccButton>
                    </slot>
                  </div>
                </div>
              </div>
            </DialogPanel>
          </div>
        </TransitionChild>
      </div>
    </Dialog>
  </TransitionRoot>
</template>
