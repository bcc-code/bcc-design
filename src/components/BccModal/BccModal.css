@layer components {
    .bcc-modal {
        @apply relative flex flex-col gap-6 rounded-lg bg-primary p-6 text-primary shadow-xl sm:min-w-[28.5rem] sm:max-w-[28.5rem] xl:max-w-[32rem] max-h-[80vh];
    }

    /* Overlay/Wrapper */
    .bcc-modal-overlay-wrapper {
        @apply fixed inset-0 z-50;
    }
    .bcc-modal-overlay {
        @apply fixed inset-0 bg-black/80 backdrop-blur-sm transition-opacity;
    }

    .bcc-modal-wrapper {
        @apply flex h-full w-full transform flex-col justify-end transition-all sm:items-center sm:justify-center p-4;
    }

    /* Header */
    .bcc-modal-header {
        @apply border-b border-on-primary pb-5;
    }

    /* Body */
    .bcc-modal-body {
        @apply flex flex-col gap-2 overflow-y-auto;
    }

    /* Title */
    .bcc-modal-title {
        @apply text-heading-medium flex justify-between items-start;
    }
    .bcc-modal-close-button {
        @apply text-black;
    }
    .bcc-modal-close-icon {
        @apply h-6 w-6;
    }

    /* Content */
    .bcc-modal-content {
        @apply text-body-base overflow-y-auto;
    }

    /* Footer */
    .bcc-modal-footer {
        @apply sticky bottom-0 left-0 right-0 bg-primary;
    }

    .bcc-modal-footer-content {
        @apply mb-5;
    }

    /* Actions */
    .bcc-modal-actions {
        @apply flex flex-col gap-4 sm:flex-row sm:justify-end sm:self-end;
    }
    .bcc-modal-secondary-action {
        @apply order-2 flex flex-col;
    }
    .bcc-modal-primary-action {
        @apply order-1 flex flex-col sm:order-3;
    }
}