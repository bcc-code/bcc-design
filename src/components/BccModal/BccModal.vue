<script setup lang="ts">
import { Dialog, DialogOverlay, TransitionChild, TransitionRoot } from "@headlessui/vue";
import { CloseIcon } from "@bcc-code/icons-vue";
import BccButton from "../BccButton/BccButton.vue";

type Props = {
  open: boolean;
  title: string;
  showClose?: boolean;
};

withDefaults(defineProps<Props>(), {
  open: false,
  showClose: true,
});

const emit = defineEmits(["close"]);
</script>

<template>
  <TransitionRoot as="template" :show="open">
    <Dialog as="div" class="fixed inset-0" @close="emit('close')">
      <div class="flex h-screen w-screen items-center justify-center overflow-hidden">
        <TransitionChild
          as="div"
          enter="ease-out duration-300"
          enter-from="opacity-0"
          enter-to="opacity-100"
          leave="ease-in duration-200"
          leave-from="opacity-100"
          leave-to="opacity-0"
        >
          <DialogOverlay class="fixed inset-0 bg-black/80 backdrop-blur-sm transition-opacity" />
        </TransitionChild>

        <TransitionChild
          as="template"
          enter="ease-out duration-300"
          enter-from="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
          enter-to="opacity-100 translate-y-0 sm:scale-100"
          leave="ease-in duration-200"
          leave-from="opacity-100 translate-y-0 sm:scale-100"
          leave-to="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
        >
          <div
            class="flex h-full w-full transform flex-col justify-end transition-all sm:items-center sm:justify-center sm:p-5"
          >
            <div
              class="flex max-w-xl flex-col gap-6 rounded-t-lg bg-primary p-6 text-primary shadow-xl sm:min-w-lg sm:rounded-lg"
            >
              <div class="flex justify-between">
                <h1 class="text-heading-base">{{ title }}</h1>
                <button
                  @click.prevent="emit('close')"
                  v-if="showClose"
                  class="hidden sm:inline-block"
                >
                  <CloseIcon class="h-6 w-6" />
                </button>
              </div>

              <div class="text-body-base">
                <slot></slot>
              </div>

              <div class="flex flex-col gap-4 sm:flex-row sm:justify-end sm:self-end">
                <div class="order-2 flex flex-col" v-if="$slots.secondaryAction">
                  <slot name="secondaryAction"></slot>
                </div>
                <div class="order-1 flex flex-col sm:order-3" v-if="$slots.primaryAction">
                  <slot name="primaryAction">
                    <BccButton @click="emit('close')"> OK </BccButton>
                  </slot>
                </div>
              </div>
            </div>
          </div>
        </TransitionChild>
      </div>
    </Dialog>
  </TransitionRoot>
</template>
