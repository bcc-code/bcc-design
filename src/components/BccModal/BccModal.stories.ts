import BccModal from "./BccModal.vue";
import BccButton from "../BccButton/BccButton.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

export default {
  title: "Components/BccModal",
  component: BccModal,
  argTypes: {
    headingContext: {
      options: ["success"],
      control: { type: "radio" },
    },
  },
} as Meta<typeof BccModal>;

const Template: StoryFn<typeof BccModal> = (args) => ({
  components: { BccModal, BccButton },
  setup() {
    return { args };
  },
  template: `
    <BccModal v-bind="args" @close="args.open = false">
      <template #heading v-if="args.slotHeading">
        {{ args.slotHeading }}
      </template>

        <template #default>
          <span v-if="args.slotDefault">
            {{ args.slotDefault }}
          </span>
          <span v-else>
            Is your email address <strong>lida@example.org</strong>?
          </span>
        </template>

        <template #secondaryAction>
          <BccButton variant="secondary" @click="args.open = false">
            {{ args.secondaryActionText || "Cancel" }}
          </BccButton>
        </template>

        <template #primaryAction>
          <BccButton @click="args.open = false">
            {{ args.primaryActionText || "Confirm" }}
          </BccButton>
        </template>
    </BccModal>

    <BccButton variant="secondary" @click="args.open=true">Open modal</BccButton>
  `,
});

/**
 * Control the content of the modal with the `title` prop and the `default` slot. Pass buttons to the `primaryAction` and `secondaryAction` slots. Set `showClose` to `false` to hide the close button on desktops. The modal emits a `close` event when the user clicks the close button or closes the modal by clicking outside of it or pressing Escape.
 */
export const Example = Template.bind({});
Example.args = {
  open: false,
  showClose: true,
  title: "Confirm email address",
  headingContext: "success",
  slotDefault: "",
  slotHeading: "",
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccModal title="Confirm email address" :open="showModal" @close="showModal = false">
  <template #default>
    Is your email address <strong>lida@example.org</strong>?
  </template>

  <template #secondaryAction>
    <BccButton variant="secondary" @click="showModal = false">Cancel</BccButton>
  </template>

  <template #primaryAction>
    <BccButton>Confirm</BccButton>
  </template>
</BccModal>
    `,
    },
  },
};

/**
 * Display an extra heading above the normal modal content. Set the `headingTitle` and `headingContext` props to control the title of the heading. Set the `heading` slot for the content of this heading. The close button is automatically hidden.
 */
export const WithExtraHeading = Template.bind({});
WithExtraHeading.args = {
  open: false,
  showClose: true,
  title: "Join the activity chat group",
  headingContext: "success",
  slotDefault: "You can join the activity chat by clicking the link below",
  headingTitle: "Success",
  slotHeading:
    "You successfully registered to activity. Don't forget to register for activities for Day 2 & Day 3",
  primaryActionText: "Open Telegram chat",
  secondaryActionText: "Close",
};
WithExtraHeading.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccModal
  title="Join the activity chat group"
  headingTitle="Success!"
  headingContext="success"
  :open="showModal"
  @close="showModal = false"
>
  <template #heading>
    You successfully registered to activity. Don't forget to register for activities for Day 2 & Day 3
  </template>

  <template #default>
    You can join the activity chat by clicking the link below
  </template>

  <template #secondaryAction>
    <BccButton variant="secondary" @click="showModal = false">Cancel</BccButton>
  </template>

  <template #primaryAction>
    <BccButton>Confirm</BccButton>
  </template>
</BccModal>
    `,
    },
  },
};
