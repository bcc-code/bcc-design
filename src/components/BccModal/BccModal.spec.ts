import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccModal from "./BccModal.vue";

describe("BccModal", () => {
  it("renders a modal", () => {
    const wrapper = mount(BccModal, {
      props: { open: true, title: "Modal title" },
      slots: {
        default: "Modal content",
        header: "Header slot",
        footer: "Footer slot`",
        primaryAction: "Primary action",
        secondaryAction: "Secondary action",
      },
      global: {
        renderStubDefaultSlot: true,
      },
      shallow: true,
    });

    expect(wrapper.text()).toContain("Header slot");
    expect(wrapper.text()).toContain("Modal title");
    expect(wrapper.text()).toContain("Modal content");
    expect(wrapper.text()).toContain("Footer slot");
    expect(wrapper.text()).toContain("Primary action");
    expect(wrapper.text()).toContain("Secondary action");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
