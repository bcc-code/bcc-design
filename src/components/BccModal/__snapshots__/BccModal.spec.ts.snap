// Vitest Snapshot v1

exports[`BccModal > renders a modal 1`] = `
"<anonymous-stub as=\\"template\\" show=\\"true\\" unmount=\\"true\\" appear=\\"false\\" enter=\\"\\" enterfrom=\\"\\" enterto=\\"\\" entered=\\"\\" leave=\\"\\" leavefrom=\\"\\" leaveto=\\"\\">
  <dialog-stub as=\\"div\\" static=\\"false\\" unmount=\\"true\\" open=\\"DC8F892D-2EBD-447C-A4C8-A03058436FF4\\" id=\\"headlessui-dialog-1\\" class=\\"fixed inset-0\\">
    <div class=\\"flex h-screen w-screen items-center justify-center overflow-hidden\\">
      <anonymous-stub as=\\"div\\" unmount=\\"true\\" appear=\\"false\\" enter=\\"ease-out duration-300\\" enterfrom=\\"opacity-0\\" enterto=\\"opacity-100\\" entered=\\"\\" leave=\\"ease-in duration-200\\" leavefrom=\\"opacity-100\\" leaveto=\\"opacity-0\\">
        <div class=\\"bcc-modal-overlay\\"></div>
      </anonymous-stub>
      <anonymous-stub as=\\"template\\" unmount=\\"true\\" appear=\\"false\\" enter=\\"ease-out duration-300\\" enterfrom=\\"opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95\\" enterto=\\"opacity-100 translate-y-0 sm:scale-100\\" entered=\\"\\" leave=\\"ease-in duration-200\\" leavefrom=\\"opacity-100 translate-y-0 sm:scale-100\\" leaveto=\\"opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95\\">
        <div class=\\"bcc-modal-wrapper\\">
          <dialog-panel-stub as=\\"div\\" id=\\"headlessui-dialog-panel-2\\" class=\\"bcc-modal\\">
            <div class=\\"bcc-modal-header\\">Header slot</div>
            <div class=\\"bcc-modal-title\\">
              <dialog-title-stub as=\\"h3\\" id=\\"headlessui-dialog-title-3\\">Modal title</dialog-title-stub>
              <!--v-if-->
            </div>
            <div class=\\"bcc-modal-content\\">Modal content</div>
            <div class=\\"bcc-modal-footer\\">
              <div class=\\"bcc-modal-footer-content\\">Footer slot\`</div>
              <div class=\\"bcc-modal-actions\\">
                <div class=\\"bcc-modal-secondary-action\\">Secondary action</div>
                <div class=\\"bcc-modal-primary-action\\">Primary action</div>
              </div>
            </div>
          </dialog-panel-stub>
        </div>
      </anonymous-stub>
    </div>
  </dialog-stub>
</anonymous-stub>"
`;
